tests/Reduce/ToAverageTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools\Tests\Reduce;
+
+use IterTools\Reduce;
+use IterTools\Tests\Fixture\ArrayIteratorFixture;
+use IterTools\Tests\Fixture\GeneratorFixture;
+use IterTools\Tests\Fixture\IteratorAggregateFixture;
+
+class ToAverageTest extends \PHPUnit\Framework\TestCase
+{
+    protected const ROUND_PRECISION = 4;
+
+    /**
+     * @test         toAverage array
+     * @dataProvider dataProviderForArray
+     * @param        array $data
+     * @param        int|float $expected
+     */
+    public function testArray(array $data, $expected)
+    {
+        // Given: $data
+
+        // When
+        $result = Reduce::toAverage($data);
+
+        $this->roundIfFloat($expected, $result);
+
+        // Then
+        $this->assertSame($expected, $result);
+    }
+
+    public function dataProviderForArray(): array
+    {
+        return [
+            //  data                    expected
+            [   [],                     null   ],
+            [   [0],                    0      ],
+            [   [null],                 0      ],
+            [   [false],                0      ],
+            [   [null, null],           0      ],
+            [   [null, false],          0      ],
+            [   [true, false],          0.5    ],
+            [   [false, true],          0.5    ],
+            [   [0, null, false],       0      ],
+            [   [1, null, false],       0.3333 ],
+            [   [1, null, true],        0.6667 ],
```

Comment:
Seeing this test case, I realize that the existing `Math` namespace functions like `runningAverage` will exhibit the same behavior. No need to make a change here now, but I think after this is merged I may go back to the Math functions and change them to throw an exception if a non-numeric value is provided, and do the same for these here. Doing math on booleans and nulls like they were legitimate numbers doesn't seem like the right thing to do. I will make that change later. Keep your code as it is as it is consistent with existing functionality for now.